### PR TITLE
TINY-6587: Review Autocompleter, Split Button, and Menu Button page

### PR DESCRIPTION
**Pull request by diogo405**
Related Ticket: 

# Autocompleter
Description of Changes:
* `classes` option of CardImage and CardText is not required. 
* move required fields to the top.
* change overview text - The idea to make it shorter is to be easier to understand [don't mention plugins, filtering and searching (these last two I'm assuming people will expect to have)].
* split examples - One for the `autocompleteitem` and other for `cardmenuitem`. 

# Split Button
Description of Changes:
* add a link to `choice of menu item` in fetch description
* change value of `value` from `string` to `any`
* remove API section from choice menu button
API is related to the primary button only.

# Menu Button
Description of Changes:
* move required fields to the top

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
Interactive examples under `_includes/live-demos/autocompleter-autocompleteitem` and `_includes/live-demos/autocompleter-cardmenuitem`
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
Deleted `_includes/live-demos/autocompleter`
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

**Review by lnewson — APPROVED** [review]

File: ui-components/autocompleter.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ keywords: autocomplete
 
 ## Overview
 
-An `autocompleter` enables users to quickly find and select from a pre-populated list of values as they type, leveraging searching and filtering. It provides suggestions to insert while the user is typing into the content. For example, with the [charmap]({{site.baseurl}}/plugins/opensource/charmap/) plugin enabled, typing **:amp** should show the ampersand item in the menu. To activate the `autocompleter`, the user must precede the trigger character (colon in this case) with a separator character, such as a space or a newline. Pressing `esc` should close the autocomplete menu.
+An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (e.g. ':'). Pressing `esc` should close the autocompleter.
```
Comment (tylerkelly13) [inline ui-components/autocompleter.md:11]:
```suggestion
An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key should close the autocompleter.
```

File: ui-components/autocompleter.md [hunk 2]
```diff
@@ -164,6 +164,8 @@ A `CardContainer` is a layout component used to apply a layout to an array of ca
 
 ## Interactive example
 
-The following shows two examples of how to create a charmap autocompleter. The first autocompleter is constructed using the standard autocompleter item and will show whenever a `:` character is typed plus at least one additional character. The second autocompleter is constructed using [CardMenuItems](#cardmenuitem) and will show whenever a `-` character is typed plus at least one additional character.
+The following shows two examples of how to create a charmap autocompleter. The first example is constructed using the standard autocompleter item and will show whenever a `:` character is typed plus at least one additional character. The second is constructed using [CardMenuItems](#cardmenuitem) and will show whenever a `-` character is typed plus at least one additional character.
```
Comment (tylerkelly13) [inline ui-components/autocompleter.md:167]:
```suggestion
The following examples show how to create a special characters autocompleter. The first example uses the standard autocompleter item and will show when user types the colon (`:`) character and at least one additional character. The second uses [CardMenuItems](#cardmenuitem) and will show when a user types a hyphen (`-`) character and at least one additional character.
```

File: ui-components/autocompleter.md [hunk 3]
```diff
@@ -164,6 +164,8 @@ A `CardContainer` is a layout component used to apply a layout to an array of ca
 
 ## Interactive example
```
Comment (tylerkelly13) [inline ui-components/autocompleter.md:165]:
```suggestion
## Interactive examples
```

File: _includes/components/choice-menu-items.md [hunk 4]
```diff
@@ -6,21 +6,10 @@ Choice menu items are a special type of menu item used for split toolbar button
...
+| value | any | required | A value that is passed to `onItemAction` when the choice menu item is clicked. |
 | text | string | optional | Text to display. |
 | icon | string | optional | {{site.predefinedIconsOnly}} |
-| value | string | required | A value that is passed to `onItemAction` when the choice menu item is clicked. |
 | disabled | boolean | optional | default: false - Represents the menu item's state. When true, the menu item is unclickable. Toggled by the menu item's API. |
 | shortcut | string | optional | Text to display in the shortcut label. To register a shortcut, see: [Add custom shortcuts to TinyMCE]({{site.baseurl}}/advanced/keyboard-shortcuts/#addcustomshortcutstotinymce). |
 
 > **Note**: The `icon` option for choice menu items was added in {{site.productname}} 5.3.
-
-#### API
```
Comment (tylerkelly13) [inline _includes/components/choice-menu-items.md:17]: Why where these removed? Are they incorrect or irrelevant?

**Review by tylerkelly13 — CHANGES_REQUESTED** [review]
Looks good :)
I've suggested some tweaks.

File: ui-components/autocompleter.md (hunk 1 above)
Comment (tylerkelly13) [reply ui-components/autocompleter.md:11]: FYI: https://www.tiny.cloud/docs/configure/contributing-docs/#keepitsimple re `e.g.`

File: _includes/components/choice-menu-items.md (hunk 4 above)
Comment (lnewson) [reply _includes/components/choice-menu-items.md:17]: We mentioned this the other day on slack, but these APIs don't exist/aren't exposed and should never have been documented :cry:

Comment (tylerkelly13) [reply _includes/components/choice-menu-items.md:17]: ah. Carry on.

**Review by tylerkelly13 — APPROVED** [review]
LGTM

File: ui-components/autocompleter.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ keywords: autocomplete
 
 ## Overview
 
-An `autocompleter` enables users to quickly find and select from a pre-populated list of values as they type, leveraging searching and filtering. It provides suggestions to insert while the user is typing into the content. For example, with the [charmap]({{site.baseurl}}/plugins/opensource/charmap/) plugin enabled, typing **:amp** should show the ampersand item in the menu. To activate the `autocompleter`, the user must precede the trigger character (colon in this case) with a separator character, such as a space or a newline. Pressing `esc` should close the autocomplete menu.
+An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key should close the autocompleter.
```
Comment (lnewson) [inline ui-components/autocompleter.md:11]: We've lost this bit of data in this rewrite, can you please make sure that remains as it's important:
> the user must precede the trigger character (colon in this case) with a separator character, such as a space or a newline.

Comment (JamesToohey) [inline ui-components/autocompleter.md:11]: `should close` or `will close`?

**Review by lnewson — APPROVED** [review]

**Review by JamesToohey — APPROVED** [review]

**Review by EkimChau — APPROVED** [review]

File: ui-components/autocompleter.md (hunk 5 above)
Comment (tylerkelly13) [reply ui-components/autocompleter.md:11]:
```suggestion
An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is typed in (such as '`:`'). Pressing the Escape key will close the autocompleter.
```

Comment (tylerkelly13) [reply ui-components/autocompleter.md:11]:
```suggestion
An autocompleter displays suggestions while the user is typing. Suggestions are shown when the trigger character is entered after a space or at the start of a new line (such as '` :`'). Pressing the Escape key should close the autocompleter.
```

Comment (diogo405) [reply ui-components/autocompleter.md]: I've changed to `will close`.

**Review by JamesToohey — APPROVED** [review]

**Pull request merged**